bot/exts/utils/snekbox/_cog.py

```diff
@@ -373,11 +382,39 @@ async def send_job(self, ctx: Context, job: EvalJob) -> Message:
 
                 msg += f"\n{Emojis.failed_file} {blocked_msg}"
 
+            # Split text files
+            text_files = [f for f in allowed if f.suffix in TXT_LIKE_FILES]
+            # Inline until budget, then upload to paste service
+            budget_lines = MAX_OUTPUT_BLOCK_LINES
+            budget_chars = MAX_OUTPUT_BLOCK_CHARS
+            for file in text_files:
+                file_text = file.content.decode("utf-8", errors="replace") or "[Empty]"
+                # Override to always allow 1 line and < 50 chars, since this is less than a link
+                if len(file_text) < 50 and file_text.count("\n") < 2:
+                    msg += f"\n`{file.name}`\n```\n{file_text}\n```"
+                # otherwise, use budget
+                else:
+                    format_text, link_text = await self.format_output(
+                        file_text,
+                        budget_lines,
+                        budget_chars,
+                        line_nums=False,
+                        mark_no_output=False
+                    )
+                    # With any link, use it (don't use budget)
+                    if link_text:
+                        msg += f"\n`{file.name}`\n{link_text}"
+                    else:
+                        msg += f"\n`{file.name}`\n```\n{format_text}\n```"
+                        budget_lines -= file_text.count("\n")
```

Comment:
Same here, the number of lines added is the count + 1

Reply:
More precisely, this should strip newlines since they won't be displayed in the message.

Reply:
Fixed in [7e7f7c3](https://github.com/python-discord/bot/pull/2326/commits/7e7f7c3ca1a0edeb51d741edbeffed50421a773a)
- Now uses count of `\n` + 1, and uses `format_text` which will have already been striped by `format_output`

```diff
- budget_lines -= file_text.count("\n")
+ budget_lines -= format_text.count("\n") + 1
```